source/javascript/src/daily.js

```diff
@@ -199,38 +199,42 @@ function getLogInfoAsJSON (cb) {
         const dateConverter = new DateConverter(Number(cursor.value.current_log))
         console.log(cursor.value)
         let match = false
+        let lenArr = 0
         cursor.value.$defs['daily-logs'].forEach((log, index) => {
           if (dateConverter.equals(Number(log.properties.date.time))) {
             match = true
             cb.bind(this)
             cb(cursor.value.$defs['daily-logs'][index])
           }
+          lenArr = index
         })
         if (!match) {
-          // TODO: creating new
-          // {
-          //   "type": "object",
-          //   "required": [ "date", "description" ],
-          //   "properties": {
-          //     "date": {
-          //       "type": "string",
-          //       "time": "",
-          //       "description": "The date of the event."
-          //     },
-          //     "events": [],
-          //     "tasks": [],
-          //     "notes": [],
-          //   "reflection": [],
-          //     "mood": {
-          //       "type": "number",
-          //       "multipleOf": 1,
-          //       "minimum": 0,
-          //       "exclusiveMaximum": 100,
-          //       "value": 50,
-          //       "description": "Daily mood on a range of 0-99."
-          //     }
-          //   }
-          // }
+          const appendObj = {
+            type: 'object',
+            required: ['date', 'description'],
+            properties: {
+              date: {
+                type: 'string',
+                time: cursor.value.current_log,
```

Comment:
This logic only works with clicking on daily log from the nav bar right? if we click on the yesterday/tomorrow chevron on daily log, it won't work?

Reply:
actually current_log is only updated when the add button on index is clicked right? so if the user didn't click the add button, and click the daily log in nav bar right away, the curren_log value would be stale

Reply:
@yuzilyu  This is a general fix to any and all times you try to access a direct log that does not exist- currently this only happens when you click on daily log currently, but when I have more time later today or tomorrow I will implement the yesterday/tm buttons. Hopefully implementing those buttons will be a straight forward fix- just need to generate UNIX timestamps for those days.

Reply:
okay I'm gonna leave this pr open for now then, when you make new commit I'll review it again and close it

Reply:
@yuzilyu as further clarification we won't need `current_log` once we change the points of navigation to use `Router.js`

Reply:
@NoahT @yuzilyu Updated- yesterday/today buttons now work. Only problem is the code climate bug- any suggestions?